### PR TITLE
Merge parts of Alexandru's work on key management

### PR DESCRIPTION
**Pull request by ljanyst**
This PR merges Alexandru's commits from the git-bug repo except for the validator, which I was not able to test successfully.

### CODE REVIEW

File: input/input.go [hunk 1]
```diff
@@ -115,10 +116,29 @@ func BugCommentEditorInput(repo repository.RepoCommon, preMessage string) (strin
...
-// BugCommentFileInput read from either from a file or from the standard input
+const identityVersionKeyTemplate = `%s
+
+# Please enter the armored key block. Lines starting with '#' will be ignored,
+# and an empty message aborts the operation.
+`
+// IdentityVersionKeyEditorInput will open the default editor in the terminal
+// with a template for the user to fill. The file is then processed to extract
+// the key.
+func IdentityVersionKeyEditorInput(repo repository.RepoCommon, preMessage string) (string, error) {
```
Comment (dawzee) [inline input/input.go:127]: You could remove the preMessage because the single call to this passes an empty string

File: repository/git.go [hunk 2]
```diff
@@ -270,19 +270,34 @@ func (repo *GitRepo) StoreTree(entries []TreeEntry) (git.Hash, error) {
...
+
+	// `git commit-tree` uses user.signingkey and gpg.program, but not commit.gpgsign.
+	// We read commit.gpgsign ourselves and simply pass -S to `git commit-tree`.
+	config := repo.LocalConfig()
+	gpgsign, err := config.ReadBool("commit.gpgsign")
+	if err != nil && err != ErrNoConfigEntry {
+		// There are more than one entries, or some other error.
+		return "", errors.Wrap(err, "failed to read local commit.gpgsign")
+	}
+	if gpgsign {
```
Comment (dawzee) [inline repository/git.go:292]: So if the user does not have commit.gpgsign set the commits are not signed (no error, no warning), is that what we want?

**Comment by ljanyst** [comment]
Thanks for the thorough review. These are the commits Alex sent to the git-bug repo as they were. I will make the updates you request.

File: repository/git.go (hunk 2 above)
Comment (ljanyst) [reply repository/git.go:292]: That actually makes sense in the context of general git-bug, but it probably does not for us. I need to think whether we should make our mods mergeable with git-bug and then how or not bother at all.

File: input/input.go (hunk 1 above)
Comment (ljanyst) [reply input/input.go:127]: It's better to add a message.

**Comment by aleb** [comment]
@ljanyst @dawzee Thanks for taking over the keys task and merging my commits.

Please note Michael suggested some changes in the "WIP refactor keys" commit. I integrated a while ago Michael's commit in the upstream PR: https://github.com/MichaelMure/git-bug/pull/376/commits Unfortunately the JSON serialisation is incompatible with what has been merged in git-ticket. Before we start using git-ticket, it would be good if we merge the latest key management changes. 

Changes have been made also to the "Allow managing keys" commit. It might work if Michael's WIP commit is squashed and then the resulting "Allow managing keys" is cherry-picked.

**Comment by aleb** [comment]
I see you made the signing mandatory. Thinking out loud: this covers only the ticket/identity changes, not the commits in the main repo.

**Comment by ljanyst** [comment]
I am aware of both. Thanks for taking the interest.

**Pull request merged**